File: app/src/main/java/me/imzack/app/ender/injector/component/PlanCreationComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.PlanCreationPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.PlanCreationActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlanCreationPresenterModule.class)
public interface PlanCreationComponent {

    void inject(PlanCreationActivity planCreationActivity);

    //TODO 为什么这里不需要Plan getPlan()？
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/PlanCreationComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.PlanCreationPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.PlanCreationActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlanCreationPresenterModule::class))
interface PlanCreationComponent {

    fun inject(planCreationActivity: PlanCreationActivity)
}
